File: gherkin/php/src/ParserException.php
<?php

declare(strict_types=1);

namespace Cucumber\Gherkin;

class ParserException extends \Exception
{
    public Location $location;

    public function __construct(string $message, ?Location $location)
    {
        $this->location = $location ?? new Location(-1, -1);
        $message = sprintf(
            "(%s:%s): %s",
            $this->location->line,
            $this->location->column,
            $message,
        );

        parent::__construct($message);
    }
}

File: gherkin/dart/lib/src/exceptions/ParserException.dart
import 'package:gherkin/language.dart';

class ParserException implements Exception
{
  final String message;

  final Location location;

  ParserException(this.message, [this.location=Location.empty]);

  @override String toString() {
    if( location.isEmpty ) {
      return message;
    }
    return '(${location.line}:${location.column}): $message';
  }
}
